flax/struct.py

```diff
@@ -104,7 +104,9 @@ class method that provides the smart constructor.
   if '_flax_dataclass' in clz.__dict__:
     return clz
 
-  data_clz = dataclasses.dataclass(frozen=True)(clz)  # type: ignore
+  if 'frozen' not in kwargs.keys():
```

Comment:
I have the feeling that we should raise an error if users passes `frozen = False` instead to keep `struct.dataclass` instances immutable.

Reply:
ok, I enforced `frozen=True` otherwise a `ValueError` is raised, and I added a test